packages/core/src/Stores/contract-trade-store.js

```diff
@@ -428,6 +429,12 @@ export default class ContractTradeStore extends BaseStore {
         return length > 0 ? applicable_contracts[length - 1] : {};
     }
 
+    get prev_contract() {
+        const applicable_contracts = this.applicable_contracts();
+        const length = this.contracts[0]?.contract_info.current_spot_time ? applicable_contracts.length : -1;
+        return length > 0 ? applicable_contracts[length - 2] : {};
```

Comment:
```suggestion
        return length >= 2 ? applicable_contracts[length - 2] : undefined;
```
1. If the length is 1 then `applicable_contracts[length - 2]` expression might cause an issue
2. IMO, it's better to return undefined instead of an empty object for a prev_contract.

Reply:
@balakrishna-deriv 
1. `> 1` will work in the same way, i'll change it. Thank you :)
2. We are using it in the same way as the `last_contract` declared right above it. If you don't mind, I really need to have it as `{}` by default just like the `last_contract` to avoid having to put` ?` marks all along the way. :)

Reply:
For 2, we can go ahead with `{}`. We can refactor it later.

Reply:
@balakrishna-deriv  I gave it some more thought and noticed that I'm already using it with `?` marks :D. Hence, I'm ok if it's `undefined`. I also made it shorter by removing the check for `length` since if an item with a specified index doesn't exist in an array, it will return `undefined` anyway :) Thank you for noticing!
<img width="221" alt="Screenshot 2024-07-23 at 1 00 35 PM" src="https://github.com/user-attachments/assets/65750388-9112-4934-81f4-c559d1be67f6">